core/src/main/scala/kafka/coordinator/GroupCoordinator.scala

```diff
@@ -454,23 +454,45 @@ class GroupCoordinator(val brokerId: Int,
     delayedOffsetStore.foreach(groupManager.store)
   }
 
+  def getPartitions(groupId: String): List[TopicPartition] = {
+      val group = groupManager.getGroup(groupId)
+      group.fold(List[TopicPartition]())(_.allOffsets.keySet.toList)
+  }
 
   def handleFetchOffsets(groupId: String,
-                         partitions: Seq[TopicPartition]): Map[TopicPartition, OffsetFetchResponse.PartitionData] = {
+                         partitions: Seq[TopicPartition],
+                         apiVersion: Short): (Map[TopicPartition, OffsetFetchResponse.PartitionData], Short) = {
+    require(apiVersion >= 1)
+
     if (!isActive.get) {
-      partitions.map { topicPartition =>
-        (topicPartition, new OffsetFetchResponse.PartitionData(OffsetFetchResponse.INVALID_OFFSET, "", Errors.GROUP_COORDINATOR_NOT_AVAILABLE.code))}.toMap
+      (apiVersion match {
```

Comment:
Good point... One option that comes to mind is to use exceptions to propagate top-level errors. This would rely on `OffsetFetchRequest.getErrorResponse` to build the response. But we don't do that for any of the other coordinator APIs, so I'd rather not make this case exceptional.

So how about this: in addition to returning the top-level error code directly in the tuple, we also use it to fill the partition-level error code. Then we don't need to pass the version into `GroupCoordinator` at all and we can let the handler in `KafkaApis` decide how to do the serialization. Basically if the top-level error code is not NONE, then ignore the partitions.